services/audit-service/src/main/java/datawave/microservice/audit/replay/ReplayController.java

```diff
@@ -0,0 +1,726 @@
+     *
+     * @param id
+     *            The audit replay id
+     * @param sendRate
+     *            The number of messages to send per second
+     * @return status, indicating whether the update was successful
+     */
+    @ApiOperation(value = "Updates an audit replay.")
+    @RolesAllowed({"Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/{id}/update", method = org.springframework.web.bind.annotation.RequestMethod.POST)
+    public String update(@ApiParam("The audit replay id") @PathVariable("id") String id,
+                    @ApiParam(value = "The number of messages to send per second", required = true) @RequestParam long sendRate, HttpServletResponse response) {
+        
+        log.info("Updating sendRate to " + sendRate + " for audit replay with id " + id);
+        
+        String resp;
+        
+        // only update if the send rate is valid
+        if (sendRate >= 0) {
+            // pull the replay status from cache to ensure it exists
+            Status status = status(id, replayProperties.isPublishEvents());
+            if (status != null) {
+                update(status, sendRate, replayProperties.isPublishEvents());
+                resp = "Updated audit replay with id " + id;
+            } else {
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                resp = "No audit replay found with id " + id;
+            }
+        } else {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            resp = "Send rate must be >= 0";
+        }
+        
+        log.info(resp);
+        return resp;
+    }
+    
+    private void update(Status status, long sendRate, boolean publishEvent) {
+        // is the replay running?
+        if (status.getState() == RUNNING) {
+            
+            // if we own it, update it. otherwise fire an event to all of the audit services
+            RunningReplay replay = runningReplays.get(status.getId());
+            if (replay != null) {
+                replay.getStatus().setSendRate(sendRate);
+                statusCache.update(replay.getStatus());
+            } else if (publishEvent) {
+                appCtx.publishEvent(new AuditReplayRemoteRequestEvent(this, busProperties.getId(), Request.update(status.getId(), sendRate)));
+            }
+        } else {
+            
+            // just update the cache if it's not running
+            status.setSendRate(sendRate);
+            statusCache.update(status);
+        }
+    }
+    
+    /**
+     * Updates all audit replays
+     *
+     * @param sendRate
+     *            The number of messages to send per second
+     * @return status, indicating the number of audit replays which were successfully updated
+     */
+    @ApiOperation(value = "Updates all audit replays.")
+    @RolesAllowed({"Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/updateAll", method = org.springframework.web.bind.annotation.RequestMethod.POST)
```

Comment:
The "PUT" verb is usually used for update/modification of a resource.

Reply:
Updated in the next commit.